lib/AST/ASTPrinter.cpp

```diff
@@ -1134,20 +1135,6 @@ class PrintAST : public ASTVisitor<PrintAST> {
       Printer.setSynthesizedTarget(Options.TransformContext->getDecl());
     }
 
-    // We want to print a newline before doc comments.  Swift code already
-    // handles this, but we need to insert it for clang doc comments when not
-    // printing other clang comments. Do it now so the printDeclPre callback
-    // happens after the newline.
-    if (Options.PrintDocumentationComments && D->hasClangNode()) {
-      auto clangNode = D->getClangNode();
-      auto clangDecl = clangNode.getAsDecl();
-      if (clangDecl &&
-          clangDecl->getASTContext().getRawCommentForAnyRedecl(clangDecl)) {
-        Printer.printNewline();
-        indent();
-      }
-    }
-
     Printer.callPrintDeclPre(D, Options.BracketOptions);
```

Comment:
This was leftover from generated interface printing normal comments between decls, which was hooking "PrintDeclPre". Not needed.